src/content/hinting.ts

```diff
@@ -396,6 +396,64 @@ interface Hintables {
     hintclasses?: string[]
 }
 
+/** A convient javascript interface to hint on specified html elements.
+  The return value is a promise resolve to the selected element,
+  or a AsyncIterator resolve to the selected elements in rapid mode.
+
+  @param elements a iterator yield html elements
+  @param option a option object. The `option.rapid` specify whether hint in rapid mode. Default value is false. The `option.callback` is execute when a hint is selected if specified.
+
+  @returns promise resolve to the selected element, or a async iterator in rapid mode.
+ */
+export function hintElements(elements: Element[], option = {}) {
+    const hintable = toHintablesArray(Array.from(elements))
+    const rapid = option["rapid"] ?? false
+    const callback = typeof option["callback"] === "function" ?
+        option["callback"] : x => x
```

Comment:
Shouldn't we provide a more useful default like trying to click on the element? Otherwise LGTM :)

Reply:
If we introduce a default action, the async iterator will be useless without specifying a callback do nothing.

pros:

* default action may be useful.
* Async iterator code is complex; callback will be easier.

cons:

* Callback is ugly; promise/async is beautiful.

How about treat `{callback: null}` as `NOP` and `{callback: undefined}` as `e => e.click()`?

Reply:
Oh, I see. I hadn't thought about that.

Maybe we should add some short examples to the docstring instead of a useful default, like `hintElements(...).then(e=>tri.dom.simulateClick())` and `for (await e of hintElements(..., {rapid: true}){tri.dom.simulateClick(e)}` ?